lib/models/LifecycleConfiguration.ts

```diff
@@ -673,6 +675,13 @@ export default class LifecycleConfiguration {
             const msg = 'Date must be in ISO 8601 format';
             return errors.InvalidArgument.customizeDescription(msg);
         }
+        const midnightRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
+            '(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T00' +
+            ':00:00(.[0-9]+)?(Z)?$');
```

Comment:
What is AWS saying if we pass it time with 1 ms after midnight, e.g.: `2023-08-23T06:15:01.001000` ?